### ISSUE TITLE
Kokkos configuration failure with generate_makefile script

### ISSUE THREAD

**Issue by ndellingwood** [kokkos/kokkos#3423]
I am unable to configure Kokkos at SHA 1fe6f1b025fc2fa52c80ab70283b52897cbc145b with Cuda versions 9.2 nor 10.1 with the configure line below. I confirmed the configure line works with 3.2.00 release as a sanity check.

## CMake line

```bash
module load sems-env sems-cmake/3.12.2 sems-cuda/9.2 sems-gcc/5.3.0 sems-git

cmake -DCMAKE_CXX_COMPILER=$HOME/kokkos/bin/nvcc_wrapper -DCMAKE_CXX_FLAGS="-O3 -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wuninitialized" -DKokkos_ENABLE_OPENMP=ON -DKokkos_ENABLE_CUDA=ON -DKokkos_ARCH_VOLTA70=ON -DKokkos_ARCH_SNB=ON -DKokkos_ENABLE_TESTS=ON -DKokkos_ENABLE_EXAMPLES=OFF -DCMAKE_VERBOSE_MAKEFILE=ON -DCMAKE_CXX_EXTENSIONS=OFF -DKokkos_CXX_STANDARD=14 -DCMAKE_BUILD_TYPE=RELEASE $HOME/kokkos
```

**Comment by ndellingwood** [comment]
Full configuration output:

```bash
-- The CXX compiler identification is GNU 5.3.0
-- Check for working CXX compiler: /ascldap/users/ndellin/kokkos/bin/nvcc_wrapper
-- Check for working CXX compiler: /ascldap/users/ndellin/kokkos/bin/nvcc_wrapper -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - failed
-- Setting policy CMP0074 to use <Package>_ROOT variables
-- The project name is: Kokkos
-- Compiler Version: 10.1.243
-- Using -std=c++14 for C++14 standard as feature
CMake Error at cmake/kokkos_test_cxx_std.cmake:53 (MESSAGE):
  Compiler NVIDIA should support cxx_std_14, but CMake reports feature not
  supported
Call Stack (most recent call first):
  cmake/kokkos_test_cxx_std.cmake:69 (kokkos_set_cxx_standard_feature)
  cmake/kokkos_tribits.cmake:205 (INCLUDE)
  CMakeLists.txt:168 (KOKKOS_SETUP_BUILD_ENVIRONMENT)


-- Configuring incomplete, errors occurred!
See also "/ascldap/users/ndellin/kokkos/testing/Cuda9_OMP/CMakeFiles/CMakeOutput.log".
See also "/ascldap/users/ndellin/kokkos/testing/Cuda9_OMP/CMakeFiles/CMakeError.log".
```

**Comment by ndellingwood** [comment]
This seems to be an issue with cmake/3.12; I swapped modules and tested with cmake/3.17 and have no issues

**Comment by ndellingwood** [comment]
For reference, with cmake/3.12, the failure occurs because `CMAKE_CXX_COMPILE_FEATURES` is empty in the check at https://github.com/kokkos/kokkos/blob/1fe6f1b025fc2fa52c80ab70283b52897cbc145b/cmake/kokkos_test_cxx_std.cmake#L52-L54

**Comment by ndellingwood** [comment]
Updated the issue title to reflect this is a problem with cmake/3.12

**Comment by crtrott** [comment]
This works with CMAKE_CXX_STANDARD set instead of KOKKOS_CXX_STANDARD.

**Comment by ndellingwood** [comment]
> This works with CMAKE_CXX_STANDARD set instead of KOKKOS_CXX_STANDARD.

@crtrott I just confirmed that works too. I need to update the generate_makefile script which was using the KOKKOS_CXX_STANDARD (and generated my cmake line I posted)

**Comment by ndellingwood** [comment]
Renamed issue to reflect this is an issue with the generate_makefile script, since it is using KOKKOS_CXX_STANDARD instead of CMAKE_CXX_STANDARD

**Comment by masterleinad** [comment]
I don't think there is all that much that we can do here except for encouraging to use `CMAKE_CXX_STANDARD` instead of `KOKKOS_CXX_STANDARD`. It might be that some relevant variables are just not populated when `CMAKE_CXX_STANDARD` is not set.

**Comment by jrmadsen** [comment]
We should just set CMAKE_CXX_STANDARD locally to the Kokkos one.

**Comment by crtrott** [comment]
I think deprecating KOKKOS_CXX_STANDARD as a user facing thing is also fine. It doesn't seem that we still need it or?

**Comment by jrmadsen** [comment]
+1 for deprecating it

**Comment by ndellingwood** [comment]
Addressed by PR #3429, closing
